src/azure-cli/azure/cli/command_modules/storage/_client_factory.py

```diff
@@ -252,7 +253,8 @@ def cf_blob_client(cli_ctx, kwargs):
         return t_blob_client.from_blob_url(blob_url=kwargs.pop('blob_url'),
                                            credential=credential,
                                            snapshot=kwargs.pop('snapshot', None),
-                                           connection_timeout=kwargs.pop('connection_timeout', None))
+                                           connection_timeout=kwargs.pop('connection_timeout', None),
+                                           transport=get_http_transport())
```

Comment:
Client like this can't take advantage of `prepare_client_kwargs_track2` defined in `azure.cli.core`, must declare `transport` explicitly.